core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -247,7 +247,7 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
     if (isTraceEnabled) {
       val requestHeader = response.request.header
       trace(s"Sending ${requestHeader.apiKey} response to client ${requestHeader.clientId} of " +
-        s"${response.responseSend.size} bytes.")
+        s"${response.responseSend.get.size} bytes.")
```

Comment:
Nice catch. However, this could be `None`, so we should probably log something different if `responseAction` is `NoOp` or `Close`.

Reply:
Already added code to handle such situation. Please review.